cardano_node_tests/tests/test_plutus_spend_build.py

```diff
@@ -1209,3 +1238,56 @@ def test_collateral_is_txin(
 
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step1)
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step2)
+
+
+@pytest.mark.testnets
+class TestNegative:
+    """Tests for txin locking using Plutus smart contracts that are expected to fail."""
+
+    @allure.link(helpers.get_vcs_link())
+    @pytest.mark.testnets
+    @hypothesis.given(redeemer_number=st.integers(min_value=-2 ^ 64, max_value=2 ^ 64))
+    @common.hypothesis_settings()
+    def test_guessing_game_pbt(
+        self,
+        cluster_lock_guessing_game: clusterlib.ClusterLib,
+        payment_addrs_lock_guessing_game: List[clusterlib.AddressRecord],
+        fund_script_guessing_game: clusterlib.TxRawOutput,
+        redeemer_number: int,
+    ):
+        """Try to spend a locked UtxO with and an invalid redeemer.
+
+        Expect failure.
+        """
+        hypothesis.assume(redeemer_number != 42)
+
+        cluster = cluster_lock_guessing_game
+        temp_template = f"{common.get_test_id(cluster)}"
+
+        redeemer_file = "redeemer_file.datum"
+        with open(redeemer_file, "w", encoding="utf-8") as outfile:
+            json.dump({"constructor": 0, "fields": [{"int": redeemer_number}]}, outfile)
```

Comment:
Would be nice to check the other possibilities to pass redeemer values as well (except maybe cbor file).
So also `{"int":redeemer_number}` and also passing `redeemer_number` in `plutus_common.PlutusOp(redeemer_value=redeemer_number, ...)` (instead of `redeemer_file=...`).

You can e.g. check if `redeemer_number` is divisible by 2, by 3 etc. and this way you can decide how you'll pass the value.